lib/IR/ChildMemSizeBasedScheduler.cpp

```diff
@@ -149,8 +151,28 @@ void ChildMemSizeBasedScheduler::orderChildNodesAndSchedule(Node *N) {
   }
 
   // Schedule the node after all its children are scheduled.
```

Comment:
Why do we need this here? We already checked that the node is not isSheduled(N);

Reply:
Added a comment. We need to perform an extra isScheduled check here, because the code below may have scheduled the current node while scheduling its children.